starlette/middleware/__init__.py

```diff
@@ -1,17 +1,38 @@
-import typing
+from typing import Any, Iterator, Protocol, Type
+
+from typing_extensions import ParamSpec
+
+from starlette.types import ASGIApp, Receive, Scope, Send
+
+P = ParamSpec("P")
+
+
+class MiddlewareClass(Protocol[P]):  # pragma: no cover
```

Comment:
Can we somehow make this private?

Reply:
Also why is there pragma here? Doesn't it get run just by being imported?

Reply:
This particular line - as well as the class body - does get run when imported. However, the empty body (`...`) of `__init__` and `__call__` does not, since those are not been called. Putting pragma at the top of the class was mere a shortcut - not to putting pragma in both of those functions.

Reply:
Makes total sense! It just confused me.

Reply:
Thanks! :)